daemon/src/connection.rs

```diff
@@ -356,9 +360,10 @@ impl Actor {
 
         let proposed_version = Version::LATEST;
         write
-            .send(wire::TakerToMaker::HelloV2 {
+            .send(wire::TakerToMaker::HelloV3 {
```

Comment:
❓ Why is this new message about sharing one's `libp2p` peer ID not sent via `libp2p`?

Reply:
Theoretically we could also say that the maker actually does not need to safe the `PeerId` because the taker always starts the protocol with the maker and we know the Cfd's id. But like this we work more towards the future, because eventually both parties will store this information.
For the maker mostly for loggin purposes and validation. For the taker to communicate with the correct maker.

Reply:
I appreciate all the explanations and context, but I'm still wondering about my original question:

>  Why is this new message about sharing one's `libp2p` peer ID not sent via `libp2p`?

Only takers who upgrade to the new version including these changes will be able to tell the maker about their `libp2p` peer ID. Why extend the legacy networking stack for the purpose of sharing this information if we can define a protocol via `libp2p` for the same purpose?

Reply:
We *can* use the `identify` protocol for knowing the peer-id, but the question how to associate this `peer-id` with a `Cfd` that was opened on the legacy connection remains. When you open the `Cfd` you don't not know the peer-id unless we know it over the old connection or are able to "look it up".

As an alternative we could somehow store a mapping from legacy identity to peer-id in the database, or we could have some kind of task that updates `Cfd`s based on connections being opened over libp2p2. I feel that these solutions are even more intrusive.

Reply:
> We can use the identify protocol for knowing the peer-id, but the question how to associate this peer-id with a Cfd that was opened on the legacy connection remains. When you open the Cfd you don't not know the peer-id unless we know it over the old connection or are able to "look it up".

This comment on the `identify` protocol proposal talks about fetching the peer ID for a particular taker ID: https://github.com/itchysats/itchysats/issues/1769#issuecomment-1092515313.

But I understand that the fastest way forward is to do something like what you have done here!